### ISSUE TITLE
make djangocms-alias and djangocms-block work 

### ISSUE THREAD

**Issue by benzkji** [benzkji/djangocms-no-versioning#10]
... with the help of @fsbraun 

- djangocms-blog will fail, as it's code is fixed on djangocms-versioning
- djangocms-alias will not recognize djangocms-no-versioning

**Comment by fsbraun** [comment]
Some thinking needed here.

**Comment by fsbraun** [comment]
I took a look again today (only 2+ years later 🤦‍♂️). And the news is overall positive:
* Both djangocms-alias and djangocms-stories (blog replacement) have gone some way to abstract away from djangocms-versioning's internals
* **But** there is one overall unsolved problem: Each package needs to know where to get the `VersionableItem` data structure from. Currently, they are directly imported from djangocms-versioning. In the future, I believe it needs  to be advertised in the versioning's `CMSConfig` object. The tricky bit is doing this without creating a race condition (versioning or no-versioning needs to look for it, but alias and stories need to get the object from versioning or no-versioning). 

I think this is solvable and will define the future pattern for distributing data structures needed for service registration.

**Comment by benzkji** [comment]
If you solve this, it will help to establish alias and block as examples of how to correctly implement "implementation agnostic versioning", if I understand correctly?

**Comment by fsbraun** [comment]
Yes, this will require changes in alias and stories (formerly known as blog).

**Comment by fsbraun** [comment]
@benzkji Would you be ok with having this package listed as a minimal demo in the django-cms docs?

**Comment by benzkji** [comment]
@fsbraun yes for sure!

**Comment by fsbraun** [comment]
See proposal here: https://github.com/django-cms/django-cms/blob/feat/add-config-contract/docs/explanation/publishing.rst#alternative-versioning-packages Let me know if there's anything you believe should be better clarified.

**Comment by benzkji** [comment]
LGTM. Hopefully I can give my own package a try soon :)

**Comment by benzkji** [comment]
So, it does work already, then, and I'll close this issue!

**Comment by fsbraun** [comment]
Once the PR is merged into the core, both djangocms-alias and stories need (small) adjustments, and then it should work. 🫰
